### PR TITLE
Clear Codec and AVC420 fixes

### LINKED ISSUE #3642: ClearCodec failed
log:
```
- Remote framebuffer format PIXEL_FORMAT_ABGR32
- clear_context_new ++++++++++++++++++++++++++ 0x00007f036c47b010
- VBarStorageCursor set to: 0 by clear_context_reset
- VBarStorageCursor set to: 0 by clear_context_reset
- Loading Dynamic Virtual Channel rdpgfx
- VBarStorageCursor set to: 0 by clear_context_reset
- clear_context_new ++++++++++++++++++++++++++ 0x00007f036101d9f0
- VBarStorageCursor set to: 0 by clear_context_reset
- VBarStorageCursor set to: 0 by clear_context_reset
- 0x00007f036101d9f0 update VBarStorage @ 0
- 0x00007f036101d9f0 update VBarStorage @ 1
- 0x00007f036101d9f0 update VBarStorage @ 2
...
- 0x00007f036101d9f0 update VBarStorage @ 88
- 0x00007f036101d9f0 update VBarStorage @ 89
- 0x00007f036101d9f0 update VBarStorage @ 90
- clear_context_new ++++++++++++++++++++++++++ 0x00007f0362e0e410
- VBarStorageCursor set to: 0 by clear_context_reset
- VBarStorageCursor set to: 0 by clear_context_reset
- clear_context_free ------------------------ 0x00007f036101d9f0
- 0x00007f0362e0e410 update VBarStorage @ 0
- 0x00007f0362e0e410 update VBarStorage @ 1
- 0x00007f0362e0e410 vBarIndex:92  clear->VBarStorageCursor:2
- 0x00007f0362e0e410 vBarEntry->count:0 != vBarHeight:45
- clear_decompress failure: -1
- context->SurfaceCommand failed with error 1359
- rdpgfx_decode failed with error 1359!
- rdpgfx_recv_wire_to_surface_1_pdu failed with error 1359!
- Error while parsing GFX cmdId: RDPGFX_CMDID_WIRETOSURFACE_1 (0x0001)
- rdpgfx_recv_pdu failed with error 1359!
- drdynvc_order_recv failed with error 1359!
- drdynvc_virtual_channel_client_thread reported an error. Error was 1359
- Network disconnect!
- Failed to check FreeRDP file descriptor
- clear_context_free ------------------------ 0x00007f0362e0e410
- clear_context_free ------------------------ 0x00007f036c47b010
```
sequence:

create clear context  0x00007f036101d9f0
udpate VBarStorage at 0,1,2 .... 90
delete clear context 0x00007f036101d9f0

create clear context  0x00007f0362e0e410
update VBarStorage at 0,1
use VBarStorage at 92  ---> wrong

The vBarIndex(92) is too large, perhaps from clear context 0x00007f036101d9f0,
while context 0x00007f036101d9f0 has been deleted...
@akallabeth

### PR DESCRIPTION
**Pull request by akallabeth**
Fixes #3642 #3527 #3279 et al

Tested with windows 8.1 and 10 as well as android

### CODE REVIEW

**Comment by akallabeth** [comment]
@mfleisz This may fix the issues you've encountered

File: libfreerdp/codec/clear.c [hunk 1]
```diff
@@ -958,20 +958,19 @@ static BOOL clear_decompress_glyph_data(CLEAR_CONTEXT* clear,
...
 		CLEAR_GLYPH_ENTRY* glyphEntry = &(clear->GlyphCache[glyphIndex]);
 		glyphEntry->count = nWidth * nHeight;
 
 		if (glyphEntry->count > glyphEntry->size)
 		{
-			UINT32* tmp;
+			BYTE* tmp;
 			glyphEntry->size = glyphEntry->count;
-			tmp = (UINT32*) realloc(glyphEntry->pixels, glyphEntry->size * 4);
+			tmp = realloc(glyphEntry->pixels, glyphEntry->size * bpp);
```
Comment (akallabeth) [inline libfreerdp/codec/clear.c]: @mfleisz You looked at the wrong end of the diff ;)

**Comment by akallabeth** [comment]
@mfleisz Refactored the changes, using an API compatible fix now.

File: libfreerdp/codec/clear.c (hunk 1 above)
Comment (mfleisz) [inline libfreerdp/codec/clear.c]: @akallabeth You are right - should not be necessary with PODs

**Comment by mfleisz** [comment]
@freerdp-bot test

File: libfreerdp/codec/rfx.c [hunk 2]
```diff
@@ -1011,6 +1011,8 @@ BOOL rfx_process_message(RFX_CONTEXT* context, const BYTE* data, UINT32 length,
 	BOOL ok = TRUE;
 	UINT16 expectedDataBlockType = WBT_FRAME_BEGIN;
 
+	region16_init(&updateRegion);
```
Comment (mfleisz) [inline libfreerdp/codec/rfx.c]: Wouldn't it be easier to just move the declaration of updateRegion into the for() loop and get rid of init/uninit here and in fail:?

Comment (bmiklautz) [inline libfreerdp/codec/rfx.c]: @mfleisz good point.

**Comment by bmiklautz** [comment]
@freerdp-bot test.

**Comment by bmiklautz** [comment]
@freerdp-bot test.

**Comment by bmiklautz** [comment]
@freerdp-bot test.

**Comment by mfleisz** [comment]
@freerdp-bot test

**Pull request merged**